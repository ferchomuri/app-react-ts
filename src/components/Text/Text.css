.normal-text {
  font-size: 13px;
  line-height: 1.5;
  font-weight: 400;
  color: #333;
}

.bold-text {
  font-weight: 700;
  color: #333;
}

.italic-text {
  font-style: italic;
}

.underline-text {
  text-decoration: underline;
}

.left {
  text-align: left;
}

.center {
  text-align: center;
}

.right {
  text-align: right;
}

.wrap-text {
  white-space: normal !important;
}