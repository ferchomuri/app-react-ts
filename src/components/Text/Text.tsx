import { TextProps } from "../../types";
import "./Text.css";

const Text: React.FC<TextProps> = (props) => {
  const { text, type, align = "left", wrap, size } = props;

  const className = `normal-text ${type || ""} ${wrap || ""}`;
  const style = { textAlign: align, fontSize: size };

  return (
    <p className={className} style={style}>
      {text}
    </p>
  );
};

export default Text;
