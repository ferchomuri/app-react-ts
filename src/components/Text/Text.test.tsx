import { render } from "@testing-library/react";
import Text from "./Text";

describe("Text", () => {
  it("should render the component", () => {
    const { container } = render(<Text text='Test' />);
    expect(container).toMatchSnapshot();
  });
});
