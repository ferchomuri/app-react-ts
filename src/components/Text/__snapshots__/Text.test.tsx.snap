// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render the component 1`] = `
<div>
  <p
    class="normal-text  "
    style="text-align: left;"
  >
    Test
  </p>
</div>
`;
